src/query/shorthand.ts

```diff
@@ -0,0 +1,88 @@
+import {Type} from 'vega-lite/src/type';
+
+import {EncodingQuery} from './encoding';
+import {SpecQuery, stack} from './spec';
+import {enumSpecShort, isEnumSpec, SHORT_ENUM_SPEC} from '../enumspec';
+
+import {getNestedEncodingPropertyChild, Property} from '../property';
+import {keys} from '../util';
+
+export function spec(specQ: SpecQuery): string {
+  const mark = enumSpecShort(specQ.mark);
+  const encodings = specQ.encodings.map(encoding)
+                        .sort()
+                        .join('|');  // sort at the end to ignore order
+  const _stack = stack(specQ);
+
+  return mark + '|' +
+      // TODO: transform
+      (_stack ? 'stack=' + _stack.offset + '|' : '') +
+      encodings;
+}
+
+export function encoding(encQ: EncodingQuery): string {
+  return enumSpecShort(encQ.channel) + ':' + fieldDefShorthand(encQ);
+}
+
+export function fieldDefShorthand(encQ: EncodingQuery): string {
```

Comment:
This should be fieldDef, not fieldDefShorthand , although you might import it as fieldDefShorthand.

The global signature should be cql.shorthand.fieldDef